data-azure-cosmos/src/main/java/io/micronaut/data/cosmos/operations/DefaultCosmosRepositoryOperations.java

```diff
@@ -253,12 +261,144 @@ public Optional<Number> executeUpdate(PreparedQuery<?, Number> preparedQuery) {
+        T entity = operation.getEntity();
+        CosmosContainer container = getContainer(persistentEntity);
+        ObjectNode item = serializeToTree(entity, Argument.of(operation.getRootEntity()));
+        CosmosItemRequestOptions options = new CosmosItemRequestOptions();
+        CosmosItemResponse<?> cosmosItemResponse = container.deleteItem(item, options);
+        if (cosmosItemResponse.getStatusCode() == HttpResponseStatus.NO_CONTENT.code()) {
+            return 1;
+        }
         return 0;
     }
 
     @Override
     public <T> Optional<Number> deleteAll(DeleteBatchOperation<T> operation) {
-        return Optional.empty();
+        RuntimePersistentEntity<T> persistentEntity = runtimeEntityRegistry.getEntity(operation.getRootEntity());
+        CosmosContainer container = getContainer(persistentEntity);
+        CosmosItemRequestOptions options = new CosmosItemRequestOptions();
+        Argument<T> arg = Argument.of(operation.getRootEntity());
+        Iterator<T> iterator = operation.iterator();
+        String partitionKeyDefinition = getPartitionKey(persistentEntity);
+        int deletedCount;
+        if (StringUtils.isEmpty(partitionKeyDefinition)) {
+            deletedCount = 0;
+            // Bulk operations delete won't work without partition key so deleting one by one document
+            while (iterator.hasNext()) {
+                T entity = iterator.next();
+                ObjectNode tree = serializeToTree(entity, arg);
+                CosmosItemResponse<?> cosmosItemResponse = container.deleteItem(tree, options);
+                if (cosmosItemResponse.getStatusCode() == HttpResponseStatus.NO_CONTENT.code()) {
+                    deletedCount++;
+                }
+            }
+        } else {
+            List<ObjectNode> items = new ArrayList<>();
+            while (iterator.hasNext()) {
+                T entity = iterator.next();
+                items.add(serializeToTree(entity, arg));
+            }
+            deletedCount = bulkDelete(container, items.iterator(), persistentEntity, Optional.empty());
+        }
+        return Optional.of(deletedCount);
+    }
+
+    @Override
+    public Optional<Number> executeDelete(PreparedQuery<?, Number> preparedQuery) {
+        boolean isRawQuery = preparedQuery.getAnnotationMetadata().stringValue(Query.class, DataMethod.META_MEMBER_RAW_QUERY).isPresent();
+        if (isRawQuery) {
+            throw new IllegalStateException("Cosmos Db does not support raw delete queries.");
+        }
+        RuntimePersistentEntity<?> persistentEntity = runtimeEntityRegistry.getEntity(preparedQuery.getRootEntity());
+        CosmosContainer container = getContainer(persistentEntity);
+        List<SqlParameter> paramList = bindParameters(preparedQuery);
+        SqlQuerySpec querySpec = new SqlQuerySpec("SELECT * " + preparedQuery.getQuery(), paramList);
```

Comment:
Would it be better just to generate the filter query with `SELECT * ` instead of doing the concat all the time?

Reply:
It looked strange having this for delete in query builder, but it's strange either way so I will move it to builder.